### PR TITLE
Booking from menu level

### PR DESCRIPTION
**Pull request by MlulekiN**

### CODE REVIEW

File: Feature/NewBookingFromTopMenu.feature [hunk 1]
```diff
@@ -0,0 +1,31 @@
+Feature: New booking from the home page top menu
```
Comment (benedictnkosi) [inline Feature/NewBookingFromTopMenu.feature]: change the feature name to just New booking

File: Feature/NewBookingFromTopMenu.feature [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+  Scenario Outline: Making A New Booking from Home Page
```
Comment (benedictnkosi) [inline Feature/NewBookingFromTopMenu.feature]: change the scenario name to just New booking

File: Feature/NewBookingFromTopMenu.feature [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+    Given The user is on the home page
+    When The user clicks on the 'BOOK NOW' top menu
+    Then User validates booking form is displayed
+
+    When The user clicks the input grid for guest information with value'<name>' and enters guest information '<guestName>'
```
Comment (benedictnkosi) [inline Feature/NewBookingFromTopMenu.feature]: this is too long and unnecessary.
The user enters <guestName>'  in field 'guest name'
do not parameters the field name

File: src/test/java/pages/RoomPage.java [hunk 4]
```diff
@@ -3,9 +3,7 @@
 import org.openqa.selenium.By;
 
 public class RoomPage {
-    public static By inputDate = By.xpath("//div[@class='hny-frm_grid']/child::span/input[@id='rooms_checkindate']");
-    public static By calendarTable = By.xpath("//div[@class='calendar-table']");
-    public static By disabledDate = By.xpath("(//div[@class='calendar-table']//td[contains(@class,'disabled')])[1]");
+
     public static By roomContainer = By.xpath("//div[contains(@class,'maghny-gd')]");
 
     public static By dateVariable(String date) {
```
Comment (benedictnkosi) [inline src/test/java/pages/RoomPage.java]: why is this in roomPage

File: src/test/java/stepDefinition/BookNowMenuLevelSteps.java [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+    }
+
+
+    @Then("User validates booking form is displayed")
+    public void user_validates_content_form_is_displayed() {
+        WebElement bookNowContentPage = driver.findElement(BookNowPage.BookingForm);
+        Assert.assertTrue(bookNowContentPage.isDisplayed());
+    }
+
+    @When("The user clicks the input grid for guest information with value{string} and enters guest information {string}")
```
Comment (benedictnkosi) [inline src/test/java/stepDefinition/BookNowMenuLevelSteps.java]: space missing value{string}

File: src/test/java/stepDefinition/BookNowMenuLevelSteps.java [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+    public void theUserClicksTheInputGridForGuestInformationWithValueName(String inputGrid, String guestInfo) {
+        WebElement guestInfoInput = driver.findElement(BookNowPage.GuestDetailInput(inputGrid));
+        guestInfoInput.click();
```
Comment (benedictnkosi) [inline src/test/java/stepDefinition/BookNowMenuLevelSteps.java]: why are you not using the seleniumAction click?

File: src/test/java/stepDefinition/BookNowMenuLevelSteps.java [hunk 7]
```diff
@@ -0,0 +1,73 @@
...
+        Assert.assertTrue(TermsConditionsChecked.isSelected());
+    }
+
+    @When("The user clicks on the book now button on the booking form page")
+    public void theUserClicksOnTheBookNowButtonOnTheBookingFormPage() {
+        SeleniumActions.click(BookNowPage.BookingFormBookNowButton, driver);
+    }
+
+    @Then("User validates booking is successful")
+    public void userValidatesBookingIsSuccessful() {
```
Comment (benedictnkosi) [inline src/test/java/stepDefinition/BookNowMenuLevelSteps.java]: why is this empty?

File: src/test/java/stepDefinition/CommonSteps.java [hunk 8]
```diff
@@ -33,9 +39,36 @@ public void the_user_has_logged_in_the_website() {
     }
 
     @When("The user clicks on the {string} top menu")
-    public void theUserClicksOnTheTopColumn(String menu) {
+    public void theUserClicksOnTheTopMenu(String menu) {
         WebElement topMenu = driver.findElement(MenuPage.topMenu(menu));
         topMenu.click();
```
Comment (benedictnkosi) [inline src/test/java/stepDefinition/CommonSteps.java:43]: selenium Actions click please

**Review by benedictnkosi — CHANGES_REQUESTED** [review]

File: Feature/NewBookingFromTopMenu.feature (hunk 1 above)
Comment (MlulekiN) [reply Feature/NewBookingFromTopMenu.feature]: Done

File: Feature/NewBookingFromTopMenu.feature (hunk 2 above)
Comment (MlulekiN) [reply Feature/NewBookingFromTopMenu.feature]: Done

File: Feature/NewBookingFromTopMenu.feature (hunk 3 above)
Comment (MlulekiN) [reply Feature/NewBookingFromTopMenu.feature]: Done

File: src/test/java/pages/RoomPage.java (hunk 4 above)
Comment (MlulekiN) [reply src/test/java/pages/RoomPage.java]: Done

File: src/test/java/stepDefinition/BookNowMenuLevelSteps.java (hunk 5 above)
Comment (MlulekiN) [reply src/test/java/stepDefinition/BookNowMenuLevelSteps.java]: Done

File: src/test/java/stepDefinition/BookNowMenuLevelSteps.java (hunk 6 above)
Comment (MlulekiN) [reply src/test/java/stepDefinition/BookNowMenuLevelSteps.java]: Done

File: src/test/java/stepDefinition/CommonSteps.java (hunk 8 above)
Comment (MlulekiN) [reply src/test/java/stepDefinition/CommonSteps.java:43]: Done

File: Feature/New_Booking.feature [hunk 9]
```diff
@@ -0,0 +1,31 @@
+Feature: New booking
+
+  Scenario Outline: Making A New Booking
+    Given The user is on the home page
+    When The user clicks on the 'BOOK NOW' top menu
+    Then User validates booking form is displayed
+
+    When The user enters 'Your Name' in field '<guestName>'
```
Comment (benedictnkosi) [inline Feature/New_Booking.feature]: fix this

File: Feature/NewBookingFromTopMenu.feature (hunk 1 above)
Comment (benedictnkosi) [inline Feature/NewBookingFromTopMenu.feature]: delete this feature

File: src/test/java/pages/BookNowPage.java [hunk 10]
```diff
@@ -0,0 +1,20 @@
...
+    public static By selectRoomRemoveButton = By.xpath("(//div[contains(@class,'select-room')]/child::button[text()='Remove'])[1]");
+    public static By roomTile = By.xpath("//div[contains(@class,'select-room')]");
+    public static By nightsMessage = By.xpath("//div[@class='hny-frm_grid']//p[@id='nights_message']");
+    public static By TermsConditionsCheckBox = By.xpath("//input[@value='accept_terms']");
+    public static By BookingFormBookNowButton = By.xpath("//button[text()='Book Now']");
+
+    public static By GuestDetailInput(String GuestInfo) {
+        return By.xpath("//div[@class='hny-frm_grid']//input[@placeholder='" + GuestInfo + "']");
+    }
+    public static By successfulBooking = By.xpath("//div[contains(@class,'booking-forms')]/child::h3[@id='thank_message']");
```
Comment (benedictnkosi) [inline src/test/java/pages/BookNowPage.java]: spelling

Comment (benedictnkosi) [reply src/test/java/pages/BookNowPage.java]: successfulBookingMessage not just successfulBooking

**Review by benedictnkosi — CHANGES_REQUESTED** [review]

File: Feature/NewBookingFromTopMenu.feature (hunk 1 above)
Comment (MlulekiN) [reply Feature/NewBookingFromTopMenu.feature]: Done

File: Feature/New_Booking.feature (hunk 9 above)
Comment (MlulekiN) [reply Feature/New_Booking.feature]: Done

File: src/test/java/pages/BookNowPage.java (hunk 10 above)
Comment (MlulekiN) [reply src/test/java/pages/BookNowPage.java]: Done

File: src/test/java/stepDefinition/BookNowMenuLevelSteps.java [hunk 11]
```diff
@@ -0,0 +1,70 @@
...
+        WebElement TermsConditionsChecked = driver.findElement(BookNowPage.TermsConditionsCheckBox);
+        Assert.assertTrue(TermsConditionsChecked.isSelected());
+    }
+
+    @When("The user clicks on the book now button on the booking form page")
+    public void theUserClicksOnTheBookNowButtonOnTheBookingFormPage() {
+        SeleniumActions.click(BookNowPage.BookingFormBookNowButton, driver);
+    }
+
+    @Then("User validates booking is successful")
```
Comment (benedictnkosi) [inline src/test/java/stepDefinition/BookNowMenuLevelSteps.java]: spelling error

File: src/test/java/stepDefinition/BookNowMenuLevelSteps.java [hunk 12]
```diff
@@ -0,0 +1,70 @@
...
+    public void userValidatesBookingIsSuccessful() {
+        SeleniumActions.wait(BookNowPage.successfulBooking, driver);
```
Comment (benedictnkosi) [inline src/test/java/stepDefinition/BookNowMenuLevelSteps.java]: spelling error

File: src/test/java/utils/SeleniumActions.java [hunk 13]
```diff
@@ -17,11 +21,37 @@ public static void click(By by, WebDriver driver) {
...
+                    Thread.sleep(3000);
+
+                }
+            } catch (InterruptedException e) {
+                e.printStackTrace();
+            }
+        }
+    }
+
+    public static void wait(By by, WebDriver driver) {
```
Comment (benedictnkosi) [inline src/test/java/utils/SeleniumActions.java]: rename this. wait doesnt tell you what its waiting for.

**Review by benedictnkosi — CHANGES_REQUESTED** [review]

File: src/test/java/utils/SeleniumActions.java (hunk 13 above)
Comment (MlulekiN) [reply src/test/java/utils/SeleniumActions.java]: Done

File: src/test/java/utils/SeleniumActions.java [hunk 14]
```diff
@@ -17,11 +21,37 @@ public static void click(By by, WebDriver driver) {
...
-                    Thread.sleep(10);
+                    Thread.sleep(3000);
                 } catch (InterruptedException ex) {
                     throw new RuntimeException(ex);
                 }
             }
         }
     }
+
+    public static void clickUntil(By toBeDisplayed, By toClick, WebDriver driver) {
```
Comment (benedictnkosi) [inline src/test/java/utils/SeleniumActions.java]: rename this, and think of other testers. a method name must be descriptive

**Review by benedictnkosi — CHANGES_REQUESTED** [review]

**Review by benedictnkosi — APPROVED** [review]

File: src/test/java/utils/SeleniumActions.java (hunk 14 above)
Comment (MlulekiN) [reply src/test/java/utils/SeleniumActions.java]: Done

**Pull request merged**